geoh5py/shared/merging/base.py

```diff
@@ -30,150 +29,73 @@ class BaseMerger(ABC):
-                "values": data,
-                "entity_type": unique_entity_types[id_][0],
-            }
-
-        return add_data_dict
-
-    @classmethod
-    def extract_data_information(cls, input_entities: list[ObjectBase]) -> tuple:
-        """
-        Extract the information from data of the input entities.
-        :param input_entities: the list of the object to extract the data from.
-        :return: a tuple containing a list of the data information,
-            the unique entity types for each data,
-            the total length of the vertices,
-            and the number of keys to use to define the entity type.
-        """
-        # extract the data from the input entities
-        id_, previous, nb_keys = 0, 0, 1
-        all_data, unique_entity_types = [], []
         for input_entity in input_entities:
-            entity_unique_entity_types = []
-
-            for data in input_entity.children:
-                if not isinstance(data, Data):
-                    continue
-                if data.name == "Visual Parameters":
+            for ind, data in enumerate(input_entity.children):
+                if (
+                    not isinstance(data, NumericData)
+                    or data.association is None
+                    or data.n_values is None
+                    or data.values is None
+                ):
                     continue
 
-                all_data.append(
-                    [
-                        data.values,
-                        (data.entity_type, data.name, id_),
-                        data.association,
-                        previous,
-                        previous + cast(int, input_entity.n_vertices),
-                    ]
+                association = data.association.name
+                label = (data.name, data.entity_type.name, association)
+                start, end = (
+                    data_count[association],
+                    data_count[association] + data.n_values,
                 )
-                entity_unique_entity_types.append((data.entity_type, data.name))
-                unique_entity_types.append((data.entity_type, data.name, id_))
-                id_ += 1
-
-            # define how to define the entity_type
-            nb_keys = cls.compute_nb_keys(entity_unique_entity_types, nb_keys)
-            previous += cast(int, input_entity.n_vertices)
 
-        # create an intermediate array
-        all_data = np.array(all_data, dtype=object)
-
-        # prepare the attributes
-        unique_entity_types = list(
-            dict.fromkeys(
-                [entity_type[:nb_keys] for entity_type in unique_entity_types]  # type: ignore
+                # Increment label in case of duplicate match
+                if label in data_dict:
+                    values = data_dict[label].values
+                    if isinstance(values, np.ndarray) and ~np.all(
+                        np.isnan(values[start:end])
```

Comment:
That's a good idea.
However, how to ensure the user want to merge his first data, not the second one.

I agree it's stupid, so we don't really have to care about it programmatically,  but that's a possible scenario. 
I would just pass a warning there "Just the first data encountered was added"